tests/integration/test_charm.py

```diff
@@ -14,13 +14,45 @@
 
 METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
 APP_NAME = METADATA["name"]
+GRAFANA_AGENT_APPLICATION_NAME = "grafana-agent"
+
+
+@pytest.fixture(scope="module")
+async def deploy_grafana_agent(ops_test: OpsTest) -> None:
+    """Deploys grafana-agent-operator.
+
+    Args:
+        ops_test: Ops test Framework.
+    """
+    assert ops_test.model
+    await ops_test.model.deploy(
+        GRAFANA_AGENT_APPLICATION_NAME,
+        application_name=GRAFANA_AGENT_APPLICATION_NAME,
+        trust=True,
+    )
 
 
 @pytest.mark.abort_on_fail
-async def test_given_build_when_deploy_on_machine_with_1_interface_than_blocked(ops_test: OpsTest):
+async def test_given_build_when_deploy_on_machine_with_1_interface_then_blocked(ops_test: OpsTest):
     charm = await ops_test.build_charm(".")
 
     await asyncio.gather(
         ops_test.model.deploy(charm, application_name=APP_NAME),
         ops_test.model.wait_for_idle(apps=[APP_NAME], status="blocked", timeout=1000),
     )
+
+
+@pytest.mark.abort_on_fail
+async def test_given_grafana_agent_deployed_when_relate_to_grafana_agent_then_status_is_active(
```

Comment:
The name of the test doesn't match with what we assert, we wait for the charm to be blocked here.